File: rb/features.rb
# Logotypes SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module LogotypesFeatures
  def self.make_feature(name)
    case name
    when "base"
      LogotypesBaseFeature.new
    when "ratelimit"
      LogotypesRatelimitFeature.new
    when "retry"
      LogotypesRetryFeature.new
    when "test"
      LogotypesTestFeature.new
    when "timeout"
      LogotypesTimeoutFeature.new
    else
      LogotypesBaseFeature.new
    end
  end
end

File: py/logotypes_sdk/features.py
# Logotypes SDK feature factory

from logotypes_sdk.feature.base_feature import LogotypesBaseFeature
from logotypes_sdk.feature.test_feature import LogotypesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: LogotypesBaseFeature(),
        "test": lambda: LogotypesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
